pydantic/mypy.py

```diff
@@ -85,6 +87,11 @@ def get_method_hook(self, fullname: str) -> Optional[Callable[[MethodContext], T
             return from_orm_callback
         return None
 
+    def get_class_decorator_hook(self, fullname: str) -> Optional[Callable[[ClassDefContext], None]]:
+        if fullname == DATACLASS_FULLNAME:
+            return dataclasses.dataclass_class_maker_callback
```

Comment:
I'm being dumb, how do this add support for the `config` argument?

Reply:
I don't think it does yet; you'd need to use a different callback that called that one. I'm assuming this is just a first step in this direction

Reply:
makes sense. Sorry was being dense.

Reply:
@samuelcolvin 
Yes, this is the first step.
This callback checks `dataclasses` behavior like `frozen`.

> I'm being dumb, how do this add support for the config argument?

Mypy checks the config argument by the type-hint `config: Type[Any]`
If we need to inspect additional features then, we should append logic which is @dmontagu  said.

Would you want to check other behavior? 
I see a list that the plugin supports configs for `BaseModel`.
https://github.com/samuelcolvin/pydantic/blob/093474ae27d9e1737d6aa76bd6b75f3918acd2e2/pydantic/mypy.py#L132-L138
However, these configs are not supported by `dataclasses`
Sorry, I'm confusing what should I check.

Reply:
I thought with this that `@dataclass(config=Config)` would fail, but now I see that this inspection is done with standard type hints.

Reply:
Thank you. It's clear.